src/languages/typescript.js

```diff
@@ -222,8 +223,14 @@ export default function(hljs) {
       {
         begin: '\\.' + hljs.IDENT_RE, relevance: 0 // hack: prevents detection of keywords after dots
       },
+      {
+        begin: '\\b' + IDENT_RE + ':', relevance: 0 // prevents detection of keywords in property name
+      },
       DECORATOR,
-      ARGUMENTS
+      ARGUMENTS,
+      {
+        begin: '(^(?!\\.$).)' + IDENT_RE + '\\(', relevance: 0 // prevents detection of keywords in method name
```

Comment:
Not sure what the intention is here since we already have line 223 which already does this?  Unless I'm misunderstanding your regex... I'm not sure what you're intending the beginning part there to be doing exactly.